bin/git-create-branch

```diff
@@ -26,10 +34,10 @@ then
 		git push $REMOTE HEAD:refs/heads/$BRANCH
 		git fetch $REMOTE
 		git checkout --track -b $BRANCH $REMOTE/$BRANCH
-	else
-		git checkout -b $BRANCH
+		exit $?
 	fi
-else
-	git checkout -b $BRANCH
 fi
 
+git checkout -b $BRANCH
+exit $?
```

Comment:
IMHO, the last `exit $?` could be eliminated. Because shell script will return with the exit code of the final command by default.